includes/phing/src/Tasks/PlatformVersionsTask.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/**
+ * Discover latest major versions of NE Platform.
+ *
+ * PHP Version 5 and 7
+ *
+ * @category BuildSystem
+ * @package  DrupalToolkit
+ * @author   DIGIT NEXTEUROPA QA <DIGIT-NEXTEUROPA-QA@ec.europa.eu>
+ * @license  https://ec.europa.eu/info/european-union-public-licence_en EUPL
+ * @link     https://github.com/ec-europa/toolkit/blob/master/includes/phing/src/Tasks/DocGeneratorTask.php
+ */
+
+namespace Phing\Toolkit\Tasks;
+
+use Project;
+use Symfony\Component\Finder\Finder;
+
+require_once 'phing/Task.php';
+
+/**
+ * A Phing task to generate an aliases.drushrc.php file.
+ *
+ * @category BuildSystem
+ * @package  DrupalToolkit
+ * @author   DIGIT NEXTEUROPA QA <DIGIT-NEXTEUROPA-QA@ec.europa.eu>
+ * @license  https://ec.europa.eu/info/european-union-public-licence_en EUPL
+ * @link     https://github.com/ec-europa/toolkit/blob/master/includes/phing/src/Tasks/DocGeneratorTask.php
+ */
+class PlatformVersionsTask extends \Task
+{
+
+    /**
+     * The major version of the current project.
+     *
+     * @var string
+     */
+    private $_majorVersion = '';
+
+    /**
+     * The version specified in the project props.
+     *
+     * @var string
+     */
+    private $_packageVersion = '';
+
+    /**
+     * The version that will be used to download the package.
+     *
+     * @var string
+     */
+    private $versionprop = '';
+
+    /**
+     * The latest version available for this package.
+     *
+     * @var string
+     */
+    private $latestprop = '';
+ 
+    /**
+     * Check github repository and retrieve all latest major versions.
+     *
+     * @return void
+     */
+    public function main()
+    {
+        // Check if all required data is present.
+        $this->checkRequirements();
+
+        // Log to screen the current value.
+        $this->log(
+            "The platform package version is set to " . $this->_packageVersion . ".",
+            Project::MSG_INFO
+        );
+
+        // Get latest version of Platform.
+        $resp=$this->callGithubReleases('https://api.github.com/repos/ec-europa/platform-dev/releases/latest');
+        $latest_version = $resp->tag_name;
+
+        // Get latest version of NE Platform.
+        $resp=$this->callGithubReleases('https://api.github.com/repos/ec-europa/platform-dev/releases');
+        foreach($resp as $object) {
+            if ($object->draft == false) {
+                $versions[$object->published_at] = $object->tag_name;
+            }
+        }
+        ksort($versions);
+
+        // Check if end-user is providing the exact version, if not just get the latest
+        // for the major provided.
+        if (in_array($this->_packageVersion, $versions)) {
+            $this->setVersionProp($this->_packageVersion);
+        }
+        else {
+            foreach($versions as $version) {
+                $temporaryGroups[substr_compare($version, $this->_majorVersion, 0, 3)] = $version;
```

Comment:
I think this is responsible for the breaking of platform version 2.3 in the toolkit pipeline:
https://webgate.ec.europa.eu/CITnet/jira/browse/MULTISITE-21403

I'm not sure how to solve this issue. It needs to be solved if we can't guarantee that the platform won't release other packages like this.

Reply:
Fixed https://github.com/ec-europa/toolkit/pull/146/commits/fb7eb1b6cb799d7351eb14fa5bbd4744e61ec8a9